expression/builtin_math.go

```diff
@@ -493,15 +494,14 @@ type floorFunctionClass struct {
 	baseFunctionClass
 }
 
-// fixFloorAndCeilType fixes the evalTp of FLOOR and CEIL.
-func fixFloorAndCeilType(arg Expression) (retTp, argTp evalTp) {
+// getEvalTp4FloorAndCeil fixes the evalTp of FLOOR and CEIL.
+func getEvalTp4FloorAndCeil(arg Expression) (retTp, argTp evalTp) {
 	fieldTp := arg.GetType()
 	switch arg.GetTypeClass() {
 	case types.ClassInt:
-		if !mysql.HasUnsignedFlag(fieldTp.Flag) {
-			retTp, argTp = tpInt, tpInt
-		} else {
-			retTp, argTp = tpDecimal, tpInt
+		retTp, argTp = tpInt, tpInt
```

Comment:
because only when the argument is mysql.TypeLonglong(no matter whether signed or unsigned) need to return NEWDECIMAL